eras/shelley/impl/src/Cardano/Ledger/Shelley/Translation.hs

```diff
@@ -0,0 +1,57 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeFamilies #-}
+
+module Cardano.Ledger.Shelley.Translation (
+  FromByronTranslationContext (..),
+  emptyFromByronTranslationContext,
+  toFromByronTranslationContext,
+)
+where
+
+import Cardano.Ledger.Core (Era, EraCrypto, TranslationContext)
+import Cardano.Ledger.Keys
+import Cardano.Ledger.Shelley.Era (ShelleyEra)
+import Cardano.Ledger.Shelley.Genesis (ShelleyGenesis (..))
+import Cardano.Ledger.Shelley.PParams (ShelleyPParams, emptyPParams)
+import Data.Map (Map)
+import qualified Data.Map as Map
+import GHC.Generics (Generic)
+import GHC.Word (Word64)
+import NoThunks.Class (NoThunks (..))
+
+-- | Required data to translate a Byron ledger into a Shelley ledger.
+data FromByronTranslationContext era = FromByronTranslationContext
+  { fbtcGenDelegs :: !(Map (KeyHash 'Genesis (EraCrypto era)) (GenDelegPair (EraCrypto era)))
+  , fbtcProtocolParams :: !(ShelleyPParams era)
+  , fbtcMaxLovelaceSupply :: !Word64
+  }
+  deriving (Eq, Show, Generic)
```

Comment:
It should not be parameterized on the `era`, because the only era this type makes sense with is Shelley. So  `PParams` should be restricted to `ShelleyEra`, otherwise it could be parameterized on other eras like `AllegraEra` and `MaryEra`.
```suggestion
data FromByronTranslationContext c = FromByronTranslationContext
  { fbtcGenDelegs :: !(Map (KeyHash 'Genesis c) (GenDelegPairc))
  , fbtcProtocolParams :: !(PParams (ShelleyEra c))
  , fbtcMaxLovelaceSupply :: !Word64
  }
  deriving (Eq, Show, Generic)
```

That was the whole reason for changes in this PR:  #3224